src/app/modules/core/components/user-profile/user-profile.component.ts

```diff
@@ -1,14 +1,152 @@
+import { MockService } from '../../services/mock.service';
+import { validateEmail, validateMobileNo } from '../../utilities/helper';
 
 @Component({
   selector: 'app-user-profile',
   templateUrl: 'user-profile.component.html',
   styleUrls: ['user-profile.component.css'],
 })
-
 export class UserProfileComponent implements OnInit {
-  constructor() {}
+  public screenScope: { profile: boolean; contactList: boolean };
+  public userProfileForm: FormGroup;
+  public filteredStates: Array<string>;
+  public filteredCities: Array<string>;
+  public stateFilter;
+  public cityFilter;
+  public masterStateCityList: Array<StateCity>;
+  public minDate: Date;
+  public maxDate: Date;
+
+  constructor(
+    private snackBar: MatSnackBar,
+    private formBuilder: FormBuilder,
+    private mockService: MockService
+  ) {
+    this.screenScope = { profile: false, contactList: true };
+    const currentYear = new Date().getFullYear();
+    this.minDate = new Date(currentYear - 20, 0, 1);
+    this.maxDate = new Date(currentYear + 1, 11, 31);
+  }
+
+  ngOnInit() {
+    this.mockService.getCities().subscribe((list) => {
+      if (Array.isArray(list) && list.length > 0) {
+        this.masterStateCityList = list;
+        this.filteredStates = Array.from(
+          new Set(
+            list.map((stateCity: StateCity) => {
+              return stateCity.state;
+            })
+          )
+        );
+      }
+    });
+    this.setDefaultValuesInForm();
+  }
+
+  private setDefaultValuesInForm() {
+    let userProfile: UserProfile;
+    let existingProfile: UserProfile = JSON.parse(
+      localStorage.getItem('myProfile')
+    );
+    let compositeField = localStorage.getItem('userId');
+    userProfile = {
+      fullName:
+        existingProfile && existingProfile.fullName
+          ? existingProfile.fullName
+          : '',
+      emailId:
+        existingProfile && existingProfile.emailId
+          ? existingProfile.emailId
+          : compositeField && validateEmail(compositeField)
+          ? compositeField
+          : '',
+      mobileNo:
+        existingProfile && existingProfile.mobileNo
+          ? existingProfile.mobileNo
+          : compositeField && validateMobileNo(compositeField)
+          ? compositeField
+          : '',
+      compositeField:
+        existingProfile && existingProfile.fullName
+          ? existingProfile.fullName
+          : '',
+      dateOfBirth:
+        existingProfile && existingProfile.dateOfBirth
+          ? existingProfile.dateOfBirth
+          : '',
+      state:
+        existingProfile && existingProfile.state ? existingProfile.state : '',
+      city: existingProfile && existingProfile.city ? existingProfile.city : '',
+    };
+    this.initializeUserProfileForm(userProfile);
+  }
+
+  private initializeUserProfileForm(userProfile: UserProfile) {
+    this.userProfileForm = this.formBuilder.group({
+      fullName: [
+        userProfile.fullName ? userProfile.fullName : '',
```

Comment:
Cant we set the details directly here, rather than setting userProfile from the above method? Both methods are doing same thing right?

Reply:
This was done to maintain the readibility

Reply:
Since you have already handled empty string check in the above method, we can directly set the fields here with out any check. What do you think?

Reply:
agree